vm/src/obj/objdict.rs

```diff
@@ -44,13 +45,28 @@ impl PyDictRef {
         kwargs: KwArgs,
         vm: &VirtualMachine,
     ) -> PyResult<PyDictRef> {
-        let mut dict = DictContentType::default();
+        let dict = DictContentType::default();
 
+        let entries = RefCell::new(dict);
+        // it's unfortunate that we can't abstract over RefCall, as we should be able to use dict
+        // directly here, but that would require generic associated types
```

Comment:
I'm not sure what you mean by this comment?

The natural thing for me for is to have merge take ```&mut DictContentType``` which is simpler and works fine. The short-lived borrows in your version are probably better though at avoiding annoying borrow errors.

Reply:
As you hinted at, the problem with merge taking `&mut DictContentType` is that for the iterable case, arbitrary python code is executed. We can't borrow the `RefCell` over any invocation of arbitrary python code as it may reference the dict we are updating, leading to a panic.

What I originally wanted to do is define merge as:
```rust
fn merge<B: BorrowMut<DictContentType>>(
        dict: B,
        dict_obj: OptionalArg<PyObjectRef>,
        kwargs: KwArgs,
        vm: &VirtualMachine,
    ) -> PyResult<()> {
    ...
}
```
So it could accept both `&mut DictContentType` and `&RefCell`, but I discovered that `RefCell` doesn't actually implement `BorrowMut`, but has it's own `borrow_mut` method.

Reply:
Ah, I see. I think I prefer the non-generic version in any case. There aren't be enough callers to justify the extra complexity.